src/python/pants/backend/shell/util_rules/shell_command.py

```diff
@@ -245,7 +245,12 @@ async def _interactive_shell_command(
     boot_script = f"cd {shlex.quote(relpath)}; " if relpath != "" else ""
 
     return Process(
-        argv=(bash.path, "-c", boot_script + command, shell_name),
+        argv=(
+            bash.path,
+            "-c",
+            boot_script + command,
+            f"{bin_name()} run {shell_command.address.spec}",
```

Comment:
I actually think we ought to also include `--` to be the most correct. It's awkward with no args, but is never wrong:

E.g.
```
$ pants run <whatever>
My Super special tool
Usage:
    Run the tool: pants run <whatever>
    Ask for help: pants run <whatever> --help
```

won't work, but:
```
$ pants run <whatever>
My Super special tool
Usage:
    Run the tool: pants run <whatever> --
    Ask for help: pants run <whatever> -- --help
```

Reply:
more correct, less pretty.. 🤷🏽

Reply:
Yeah :unamused: